execution-engine/src/fs.rs

```diff
@@ -270,9 +317,10 @@ impl InodeImpl {
 
     /// Return the number of the bytes, if it is a file,
     /// or the number of inodes, if it it is a directory.
+    #[inline]
     pub(self) fn len(&self) -> FileSystemResult<FileSize> {
         let rst = match self {
-            Self::File(f) => f.len(),
+            Self::NativeModule(.., f) | Self::File(f) => f.len(),
             Self::Directory(f) => f.len(),
```

Comment:
Out of curiosity, why is this `match` statement even necessary? Since each block is doing the same thing (calling `f.len()`), why can't we just call `f.len()` without the match? Is it something weird with enums?

Reply:
The `f` comes from the `match` in each case (i.e. it's not the same `f` in each branch of the `match`).